psi4/driver/procrouting/response/scf_response.py

```diff
@@ -255,165 +261,398 @@ def _print_output(complete_dict, output):
-    else:
-        engine = TDUSCFEngine(wfn, ptype=ptype)
-
-    # just energies for now
-    solver_results = []
+    engine = TDRSCFEngine(wfn, ptype=ptype.lower(), triplet=True if spin_mult == "triplet" else
+                          False) if restricted else TDUSCFEngine(wfn, ptype=ptype.lower())
+    irrep_GS = wfn.molecule().irrep_labels()[engine.G_gs]
     for state_sym, nstates in enumerate(states_per_irrep):
         if nstates == 0:
             continue
+        irrep_ES = wfn.molecule().irrep_labels()[state_sym]
+        core.print_out(f"\n\n  ==> Seeking the lowest {nstates} {spin_mult} states with {irrep_ES} symmetry")
         engine.reset_for_state_symm(state_sym)
-        guess_ = engine.generate_guess(nstates * 2)
-
-        vecs_per_root = max_ss_vec // nstates
-
-        # ret = (ee, rvecs, stats) (TDA)
-        # ret = (ee, rvecs, lvecs, stats) (full TDSCF)
-        ret = solve_function(
-            engine=engine,
-            e_tol=e_tol,
-            r_tol=r_tol,
-            max_vecs_per_root=vecs_per_root,
-            nroot=nstates,
-            guess=guess_,
-            verbose=verbose)
-
-        # store excitation energies tagged with final state symmetry (for printing)
-        # TODO: handle R eigvecs (TDA) R/L eigvecs(full TDSCF): solver maybe should return dicts
-        for ee in ret[0]:
-            solver_results.append((ee, state_sym))
-
-    # sort by energy symmetry is just meta data
-    solver_results.sort(key=lambda x: x[0])
-
-    # print excitation energies
-    core.print_out("\n\nFinal Energetic Summary:\n")
-    core.print_out("        " + (" " * 20) + " " + "Excitation Energy".center(31) + " {:^15}\n".format("Total Energy"))
-    core.print_out("    {:^4} {:^20} {:^15} {:^15} {:^15}\n".format("#", "Sym: GS->ES (Trans)", "[au]", "[eV]",
-                                                                    "(au)"))
-    core.print_out("    {:->4} {:->20} {:->15} {:->15} {:->15}\n".format("-", "-", "-", "-", "-"))
+        guess_ = engine.generate_guess(nstates * 4)
+
+        # ret = {"eigvals": ee, "eigvecs": (rvecs, rvecs), "stats": stats} (TDA)
+        # ret = {"eigvals": ee, "eigvecs": (rvecs, lvecs), "stats": stats} (RPA)
+        ret = solve_function(engine, nstates, guess_, maxiter)
+
+        # check whether all roots converged
+        if not ret["stats"][-1]["done"]:
+            # raise error
+            raise TDSCFConvergenceError(maxiter, wfn, f"singlet excitations in irrep {irrep_ES}", ret["stats"][-1])
+
+        # flatten dictionary: helps with sorting by energy
+        # also append state symmetry to return value
+        for e, (R, L) in zip(ret["eigvals"], ret["eigvecs"]):
+            irrep_trans = wfn.molecule().irrep_labels()[engine.G_gs ^ state_sym]
+
+            # length-gauge electric dipole transition moment
+            edtm_length = engine.residue(R, mints.so_dipole())
+            # lenght-gauge oscillator strength
```

Comment:
lenght typo

Reply:
I really can't write this word without making a mistake...